src/meta-srv/src/procedure/region_migration.rs

```diff
@@ -91,7 +92,9 @@ impl PersistentContext {
         let lock_key = vec![
             CatalogLock::Read(&self.catalog).into(),
             SchemaLock::read(&self.catalog, &self.schema).into(),
-            TableLock::Read(region_id.table_id()).into(),
+            // The optimistic updating of table route is not working very well,
+            // so we need to use the write lock here.
+            TableLock::Write(region_id.table_id()).into(),
```

Comment:
This is a big change that will prevent all regions belonging to the same table from concurrently executing migrations. 🤔

Reply:
We can switch it to write lock and see if it works well.